activerecord/test/cases/calculations_test.rb

```diff
@@ -993,11 +993,24 @@ def test_pluck_functions_with_alias
   end
 
   def test_pluck_functions_without_alias
-    assert_equal [
-      [1, "The First Topic"], [2, "The Second Topic of the day"],
-      [3, "The Third Topic of the day"], [4, "The Fourth Topic of the day"],
-      [5, "The Fifth Topic of the day"]
-    ], Topic.order(:id).pluck(
+    expected = if current_adapter?(:PostgreSQLAdapter)
+      # Postgres returns the same name for each column in the given query, so each column is named "coalesce"
+      # As a result Rails cannot accurately type cast each value.
+      # To work around this, you should use aliases in your select statement (see test_pluck_functions_with_alias).
```

Comment:
This is a bit yucky, but it only happens in very specific cases: if you have select two (or more) columns that don't have an alias and use the same SQL function (eg COALESCE). This happens because of how Postgres returns column names, see https://github.com/rails/rails/pull/36186#issuecomment-618418281 for a summary.

As far as I can tell we have a few options:

- The current state, which results in this bug: https://github.com/rails/rails/issues/41651
- This PR, which fixes https://github.com/rails/rails/issues/41651 but introduces this inconsistency between databases.
- Implement the refactor in https://github.com/rails/rails/pull/36186, which adds a fair bit of overhead to every query for a relatively niche issue.

In my opinion this PR is the least bad option.